Prepare release letter for 20200921 (#81)
exasol/script-languages-release: doc/changes/changes-20200803.md

# script-languages-release 20200803, released 2020-08-13
 
## Summary
This release introduces the new standard-flavor for Exasol 7.0+. Additionally, it adds new packages to the standard-flavors, such as boto3, pysftp and paramiko and fixes several bugs for the UDFs, the script-language containers and exaslct. 

However, most importantly, it greatly simplifies the customization of the script-language containers. Each flavor now provides package lists under the directory `flavors/<flavor-name>/flavor_customization/packages`. Addtional packages entered into these lists get automatically installed when building the flavor. Furthormore, we provide detailed instrunctions how to do other kinds of modifications to the container by changing the Dockerfile at `flavors/<flavor-name>/flavor_customization/Dockerfile`. Please find more information about this, in our [Readme](https://github.com/exasol/script-languages-release#how-to-customize-an-existing-flavor).
 
## Bug Fixes

* UDFs:
  * [script-languages-release@#67](https://github.com/exasol/script-languages-release/issues/67): Fix error in Python3 UDFs when emitting only a dataframe with one Timestamp column 

>   * [script-languages-release@#48](https://github.com/exasol/script-languages-release/pull/48): Fix Python3 UDF bug with reset after get_dataframe reaches the end of the iterator

Author: yes, obvious a type, but the correction should be:

`Fix Python3 UDF bug during reset after get_dataframe reaches the end of the iterator`

  * [script-languages-release@#26](https://github.com/exasol/script-languages-release/issues/26): Add boto3 to standard flavor 
  * [script-languages-release@#25](https://github.com/exasol/script-languages-release/issues/25): Add pysftp and paramiko to standard flavor
  * [script-languages-release@#32](https://github.com/exasol/script-languages-release/issues/32): Add exasol websocket-api for Python3 to standard flavor
  * [script-languages-release@#57](https://github.com/exasol/script-languages-release/issues/57): Simplify the customization of flavors for users
  * [script-languages-release@#23](https://github.com/exasol/script-languages-release/issues/23): Add script to install specific version of a R packages
  * [script-languages@#127](https://github.com/exasol/script-languages/pull/127): Add more robust package install scripts which support package versions
* Exslsct:
  * [script-languages-release@#42](https://github.com/exasol/script-languages-release/pull/42): Add command to generate language activation statement

## Refactorings

* Exaslct:

>   * [script-languages-release@#46](https://github.com/exasol/script-languages-release/pull/46): Reorganizing exaslct and extracting the test environment to [exasol/integration-test-docker-environment](https://github.com/exasol/integration-test-docker-environment)

Reviewer: ```suggestion
  * [script-languages-release@#46](https://github.com/exasol/script-languages-release/pull/46): Reorganize exaslct and extract the test environment to [exasol/integration-test-docker-environment](https://github.com/exasol/integration-test-docker-environment)
```

>   * [script-languages@#108](https://github.com/exasol/script-languages/pull/108): Add cleanup method to exaslct tasks and scripts to run exaslct tests

Reviewer: ```suggestion
  * [script-languages@#108](https://github.com/exasol/script-languages/pull/108): Add a cleanup method to exaslct tasks and scripts to run exaslct tests
```

>   * [script-languages@#98](https://github.com/exasol/script-languages/pull/98): Pass docker socket to test container

Reviewer: ```suggestion
  * [script-languages@#98](https://github.com/exasol/script-languages/pull/98): Pass a docker socket to the test container
```

Author: `Pass the docker socket to the test container`

>   * [script-languages@#142](https://github.com/exasol/script-languages/pull/142): Extract and refactored functionalities for performance testing into exatest 

Reviewer: ```suggestion
  * [script-languages@#142](https://github.com/exasol/script-languages/pull/142): Extract and refactor functionalities for performance testing into exatest 
```

>   * [script-languages@#105](https://github.com/exasol/script-languages/pull/105): Fix reference to luigis FrozenOrderedDict and pin dependency versions in pipfile

Reviewer: ```suggestion
  * [script-languages@#105](https://github.com/exasol/script-languages/pull/105): Fix a reference to luigis FrozenOrderedDict and pin dependency versions in pipfile
```
